sys/auto_init/saul/auto_init_lis3dh.c

```diff
@@ -49,18 +54,17 @@ extern saul_driver_t lis3dh_saul_driver;
 
 void auto_init_lis3dh(void)
 {
-    for (unsigned int i = 0; i < LIS3DH_NUM; i++) {
-        int res;
+    assert(LIS3DH_NUM == LIS3DH_INFO_NUM);
 
+    for (unsigned int i = 0; i < LIS3DH_NUM; i++) {
         LOG_DEBUG("[auto_init_saul] initializing lis3dh #%u\n", i);
 
-        res = lis3dh_init(&lis3dh_devs[i], &lis3dh_params[i]);
-        if (res < 0) {
+        if (lis3dh_init(&lis3dh_devs[i], &lis3dh_params[i]) < 0) {
             LOG_ERROR("[auto_init_saul] error initializing lis3dh #%u\n", i);
```

Comment:
I would prefer printing res in the error message instead of removing the variable

Reply:
Alright